src/cbmc/cbmc_parse_options.cpp

```diff
@@ -662,6 +662,8 @@ int cbmc_parse_optionst::get_goto_program(
     if(!binaries.empty())
       config.set_from_symbol_table(symbol_table);
 
+    config.set_addr_bits(cmdline, symbol_table);
```

Comment:
My apologies, yes, it is a two-step process in case goto binaries are being read. But then shouldn't those two same steps suffice to adjust the address bits?

Your suggestions go further to eliminate even that second step, or rather avoid the redundancy across tools. Indeed `read_object_and_link` could be a good place to invoke `config.set_from_symbol_table`.